### PR TITLE
Update the documentation according to agnostic traits

### PR DESCRIPTION
**Pull request by coreggon11**
Updated the old documentation to be actual we a new implementation.
Added the documentation and the code with example how to manage the structure of the project

### CODE REVIEW

**Comment by coreggon11** [comment]
Most of the project structure is done and also described in the docs, though it will need to be refactored after @xgreenx 's changes :smiley: Some methods need to be implemented in the lending contract, they are already defined along with the steps which should be programmed so it would work properly. Documentation is done for almost everything that is done (structure, individual contracts, etc.). And also it will need to be tested with integration tests, and then maybe some bugs will come up :smiley:  so they need to be fixed as well. Raise my child well.

**Comment by coreggon11** [comment]
Ok, so everything should be implemented, along with docs, I have described three core functions in docs, the rest would be the same so it is not necessary IMO. So we just need integration tests and then refactor :smiley: And review :smiley:

**Review by yarikbratashchuk — APPROVED** [review]
lgtm

File: docs/docs/smart-contracts/example/loan.md [hunk 1]
```diff
@@ -0,0 +1,327 @@
...
+    #[cfg(not(feature = "ink-as-dependency"))]
+    use ink_storage::collections::HashMap as StorageHashMap;
+    use lending_project::traits::loan::*;
+```
+You can notice that we marked some imports with `#[cfg(not(feature = "ink-as-dependency"))]`.
+It is needed to remove rust's warnings when we will import `LoanContract` as `ink-as-dependency`
+into `LendingContract` for instantiation. When we are importing some contracts as
+`ink-as-dependency`, we only import the signature of methods without
+the implementation of them. It is why the imports are not used during
+`ink-as-dependency` and it is why the rust is throwing warnings.
```
Comment (PierreOssun) [inline docs/docs/smart-contracts/example/loan.md:146]:
```suggestion
`ink-as-dependency` and it is why Rust is throwing warnings.
```

File: docs/docs/smart-contracts/example/setup_project.md [hunk 2]
```diff
@@ -0,0 +1,78 @@
+---
+sidebar_position: 2
+title: Setup the project
+---
+
+In the first step, we will define a structure of the project.
```
Comment (PierreOssun) [inline docs/docs/smart-contracts/example/setup_project.md:6]:
```suggestion
In the first step, we will define the structure of the project.
```

File: docs/docs/smart-contracts/example/setup_project.md [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+We suggest using that structure during development because it:
```
Comment (PierreOssun) [inline docs/docs/smart-contracts/example/setup_project.md:7]:
```suggestion
We suggest using that structure during development for the following reasons:
```

File: docs/docs/smart-contracts/example/setup_project.md [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+- The interface of the contracts is defined separately from the contracts. That allows others to communicate with these contracts without knowledge about the implementation and these interfaces easily can be imported to another project(that allows others to communicate with these contracts).
```
Comment (PierreOssun) [inline docs/docs/smart-contracts/example/setup_project.md:8]:
```suggestion
- The interface of the contracts is defined separately from the contracts. That allows others to communicate with these contracts without knowledge about the implementation and these interfaces can easily be imported to another project(that allows others to communicate with these contracts).
```

File: docs/docs/smart-contracts/example/setup_project.md [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+- Resolves the problem with cyclic dependencies across the project. To call the methods of the contract from the project you enough to have an interface.
+- The usage of the `ink-as-dependency` feature is minimized. That can resolve a lot of headaches in the future.
+- The implementation of big contracts can be split into small parts to simplify the development.
+- The body of the contract doesn't contain the whole implementation of the contract. That improves the readability of the contracts.
+
+The project will contain next directories:
```
Comment (PierreOssun) [inline docs/docs/smart-contracts/example/setup_project.md:14]:
```suggestion
The project will contain the following directories:
```

File: docs/docs/smart-contracts/example/setup_project.md [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+- `traits` - contains all traits(interfaces) of the contracts developed in the project.
+Traits describe the functionality of each contract and allow to do cross-contracts calls
+without knowledge about the implementation(you don't need to import the contract, you enough to use a trait).
```
Comment (PierreOssun) [inline docs/docs/smart-contracts/example/setup_project.md:17]:
```suggestion
without knowledge about the implementation(no need to import the contract, using a trait is enough).
```

File: docs/docs/smart-contracts/example/setup_project.md [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+- `impls` - contains the implementations of traits for the contracts. 
+If the contract contains several simple functions, better to implement 
+them in the body of the contract. But if the contract contains a lot of logic
+and methods, better to move(and maybe split on parts) the implementation to that directory.
+Better to store the implementation of one contract in its own directory and not mix it with others.
+- `derive` - is optional directory. OpenBrush provides the [macro](https://github.com/Supercolony-net/openbrush-contracts/blob/main/utils/brush_derive/lib.rs#L11) 
+to define a procedure derive macro of the storage trait. 
+That derive can be used to implement the storage trait for the data structure 
+in two lines(elaborate more about it later).
```
Comment (PierreOssun) [inline docs/docs/smart-contracts/example/setup_project.md:26]:
```suggestion
in two lines(more details about it later). 
```

File: docs/docs/smart-contracts/example/setup_project.md [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+(procedure macros in the rust requires a separate crate).
+- `contracts` - contains the bodies of the contracts. Each contract should be defined 
+in its own crate(it is a rule of the ink!). Each folder in that directory is a 
+crate(contract). These contracts can have the implementation inside themselves 
+or they can import the implementation from `impls`.
+
+In that structure `traits`, `impls`, and `derive` directories are the parts of on `PROJECT_NAME` crate.
+Each contract in the `contracts` directory imports the crate `PROJECT_NAME` and use it inside.
+
+Based on the rules above the structure of that example will be next:
```
Comment (PierreOssun) [inline docs/docs/smart-contracts/example/setup_project.md:38]:
```suggestion
Based on the rules above the structure will look like the following:
```

**Pull request merged**